lib/routes/account.js

```diff
@@ -1043,16 +1128,18 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified ||
-            Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
-          return reply({})
-        }
-        customs.check(
-          request.app.clientAddress,
-          sessionToken.email,
-          'recoveryEmailResendCode')
-          .then(
-            mailer.sendVerifyCode.bind(
+
+        if (!sessionToken.emailVerified) {
+          // Send email verification
+          if (Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
+            return reply({})
+          }
+
+          return customs.check(
+            request.app.clientAddress,
+            sessionToken.email,
+            'recoveryEmailResendCode')
+            .then(mailer.sendVerifyCode.bind(
               mailer,
               sessionToken,
               sessionToken.emailCode,
```

Comment:
There's a tricky edge-case here that I don't think we're handling:

1. Create an account, but don't verify it.  Throw away that sessionToken, and delete the verification email.
2. In a separate browser, login to the account.
3. See the "you must verify your account" error message, hit "reset email".
4. Receive the verification email, click through it

Expected result:  I should be able to sync using the sessionToken from step (2).

Actual result:  If I understand the flow here correctly, the email loop will have verified the original token from step (1) rather than my new token from step (2).  So If I try to sync, I'll get *another* error about having to confirm my login, and have to do the email loop again.

ISTM that while we should indeed send the account verification email here, we should have it include `sessionToken.tokenVerificationId` rather than `sessionToken.emailCode`.  That way, clicking through the email verifies whatever session I was using to trigger the email, rather than always verifying my initial session.